### ISSUE TITLE
Multiple vaccination entries on UK NHS certificate - but one QR code only

### ISSUE THREAD

**Issue by david-inno** [corona-warn-app/cwa-wishlist#567]
Hi all,

It is really great to see that the NHS vaccination certificate is now supported by the CWA. There is, however, a problem with people who are fully vaccinated: instead of being issued one QR code for each jab, the NHS issues a single QR code which contains information for both jabs. Upon scanning of the UK NHS QR code of a fully vaccinated person, the following error message appears: 

`This vaccination certificate is not supported in your app version yet. Please update your app or contact the technical hotline under "App Information". (VC_MULTIPLE_VACCINATION_ENTRIES)'

For a single-jabbed person, the certificate is properly processed and loaded onto the CWA.

Cheers,
David-Benjamin

**Comment by thomasaugsten** [comment]
We have to wait until UK public key is added to the EU gateway and they align to one certificate in a QR code

**Comment by MikeMcC399** [comment]
@thomasaugsten 
> We have to wait until UK public key is added to the EU gateway and they align to one certificate in a QR code

Is the UK participating? They are not part of the EU any more (Brexit) and they are not listed on 
https://ec.europa.eu/info/live-work-travel-eu/coronavirus-response/safe-covid-19-vaccines-europeans/eu-digital-covid-certificate_en in the section for non-EU countries. Maybe there is some exception for UK?

**Comment by thomasaugsten** [comment]
They are not participating in the e-health group but there are discussion with non eu countries to participate

**Comment by Ein-Tim** [comment]
Any updates on this?

**Comment by thomasaugsten** [comment]
CH is participating and is accepting DCC and you can convert DCC in swiss certificates
https://bag-coronavirus.ch/zertifikat/wo-kommt-das-covid-zertifikat-zum-einsatz/

**Comment by Ein-Tim** [comment]
@thomasaugsten 

Thanks, and what about NHS QR codes?

**Comment by thomasaugsten** [comment]
No update yet

**Comment by Ein-Tim** [comment]
[@coronawarnapp](https://twitter.com/coronawarnapp) [tweeted](https://twitter.com/coronawarnapp/status/1445013055713198089?s=20): 

> So far, the UK certificates are not accepted by and in the EU.
The talks are almost finished and should be accepted in the next few days.
If that is the case you could scan them immediately in the CWA without error.

And [@UKinGermany](https://twitter.com/UKinGermany) [answered](https://twitter.com/UKinGermany/status/1445017888549707778?s=20) to the tweet:

> [@coronawarnapp](https://twitter.com/coronawarnapp) – an equivalency decision between the EU and UK apps is outstanding but please note this advice from the German embassy in London: “NHS vaccination certificates are recognised in Germany if the vaccination was carried out with a vaccine approved in Germany. […] Evidence of vaccination can be provided digitally (NHS app or NHS website) or as a printout (this can be requested from the NHS by calling 119). However, a screenshot or photo is not sufficient. In addition, the final jab must have been at least 14 days […] https://uk.diplo.de/uk-en/02/coronavirus?openAccordionId=item-2467132-7-panel

**Comment by dsarkar** [comment]
@Ein-Tim Thanks!

**Comment by vaubaehn** [comment]
@dsarkar 

> If that is the case you could scan them immediately in the CWA without error.

Does the dev team know this? Unless neither the NHS DCC format has been changed to allow only one single entry, nor CWA and business rules have been adapted to accept DCCs with multiple entries, this is not true, unfortunately.

**Comment by thomasaugsten** [comment]
UK aligned to the EU standard to integrate only one cert see demo qr codes
https://github.com/eu-digital-green-certificates/dcc-quality-assurance/tree/main/GB/1.3.0

**Comment by Ein-Tim** [comment]
Some users on Twitter are reporting that this works now, they had to download a new certificate from some portal. Could you check @chopin1810?

**Comment by david-inno** [comment]
@Ein-Tim Yeah, there was a BBC article about this recently. Let me download some fresh certificates and try. I will report here.

**Comment by MikeMcC399** [comment]
The United Kingdom is now officially listed on https://ec.europa.eu/info/live-work-travel-eu/coronavirus-response/safe-covid-19-vaccines-europeans/eu-digital-covid-certificate_en under the heading:

"**Recognition of COVID certificates from third (non-EU) countries**

So far, 24 non-EU countries (and territories) joined the EU Digital COVID Certificate system. This includes:

...
United Kingdom
...

The COVID Certificates issued in those 24 countries (and territories) are accepted in the EU under the same conditions as the EU Digital COVID Certificate. Likewise, the EU Digital COVID Certificate is accepted by those 24 countries."

**Comment by david-inno** [comment]
So, I downloaded two newly generated QR codes (expiring 24/12/2021). Scanning the first one works fine, checking validity fails though (CWA 2.14.1).
However, after scanning the second code, I see an error message on the screen which reads 'The certificate is already registered in your app. (VC_ALREADY_REGISTERED)'. Same happens after fully reinstalling CWA.

**Comment by MikeMcC399** [comment]
@chopin1810 

I'm assuming from your original wishlist description that you now have two certificates and one is for your first jab with a second one for your second jab. Can you confirm?

What were the details of the failed validity check?

Were you checking the validity of a 2/2 vaccination certificate i.e. 2nd in a series of 2 jabs? or were you checking a 1/2 vaccination certificate? A 1/2 certificate should fail the validity check. If you check against rules for Germany you should get the error "The vaccination schedule must be complete (e.g., 1/1, 2/2)"

Are you using an iOS or Android device?

**Comment by MikeMcC399** [comment]
PS
@chopin1810 
If you need to remove a certificate after you have scanned it in, there are instructions for this on https://www.coronawarn.app/en/faq/#eu_dcc_remove. You should not need to resort to uninstalling the app and reinstalling it.

**Comment by ucmichan** [comment]
> So, I downloaded two newly generated QR codes (expiring 24/12/2021). Scanning the first one works fine, checking validity fails though (CWA 2.14.1). However, after scanning the second code, I see an error message on the screen which reads 'The certificate is already registered in your app. (VC_ALREADY_REGISTERED)'. Same happens after fully reinstalling CWA.

@chopin1810 
Thanks. We received further reports from users that weren't able to scan their NHS certificate. How does it work in the UK to get a new certificate? Is there a way for everybody to obtain a new certificate in the correct format? Can you tell us which portal you've used to download the new qr codes?

**Comment by MikeMcC399** [comment]
@ucmichan 
I don't have first-hand experience of UK NHS certificates, but there is some information on https://www.nhsx.nhs.uk/covid-19-response/eu-gateway/ which may help you.

**Comment by Ein-Tim** [comment]
@ucmichan 

If I'm not mistaken you are a member of the CWA-Team, right? If so, could you maybe [make your membership of the CWA organization public](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-your-membership-in-organizations/publicizing-or-hiding-organization-membership)? 

This helps new/one time contributes to differentiate between the community and members of the development team. 

Thank you!
